assets/javascripts/kitten/components/cards/similar-projects-card.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+import { ProjectSimilarCard } from 'kitten/components/cards/project-similar-card'
+
+export class SimilarProjectsCard extends Component {
+  constructor() {
+    super()
+
+    this.state = {
+      currentIndex: 0,
+    }
+  }
+
+  curryHandleArrowClick(newCurrentIndex, callback) {
+    return () => {
+      this.setState({ currentIndex: newCurrentIndex })
+      callback()
+    }
+  }
+
+  hasProjects() {
+    return this.props.projects.length > 0
+  }
+
+  isLastProjectCurrent() {
+    if (!this.hasProjects()) return false
+
+    return this.state.currentIndex == this.props.projects.length - 1
+  }
+
+  isFirstProjectCurrent() {
+    if (!this.hasProjects()) return false
+
+    return this.state.currentIndex == 0
+  }
+
+  currentProject() {
+    if (!this.hasProjects()) return
+
+    return this.props.projects[this.state.currentIndex]
+  }
+
+  currentProjectProps() {
+    if (!this.currentProject()) return
+
+    return {
+      imageSrc: this.currentProject().imageSrc,
+      title: this.currentProject().title,
+      paragraph: this.currentProject().description,
+      tags: this.currentProject().tags,
+      infos: this.currentProject().infos,
+    }
+  }
+
+  currentStep() {
+    if (!this.currentProject()) return ''
```

Comment:
C'est pas mieux de tester avec `!this.hasProjects()` plutôt ? `currentProject()` retourne le projet courant dont tu n'as pas besoin ici.